### PR TITLE
Remove bower and upgrade dependencies

### PR DESCRIPTION
**Pull request by MuriloDalRi**
* Remove bower as a dependency and use yarn to manage our frontend dependencies
* Move all our bower dependencies to package.json and update them
* Delete bootstrap.sh and run_tests.sh scripts in favour of the makefile
* Bring node version into line with production
* Update makefile for new yarn commands
* Update README with yarn instructions

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -43,35 +43,23 @@ Install new Python dependencies with pip
 
 ```make requirements-dev```
 
-Install frontend dependencies with npm and gulp
+Install frontend dependencies with yarn and gulp
 
 ```
-npm install
+yarn
```
Comment (TheDoubleK) [inline README.md]: Following the instructions here doesn't work for me as I don't have yarn installed - the readme probably needs something about how to install yarn in the first place?

Comment (MuriloDalRi) [reply README.md]: Oops missed that bit! You can install it with brew on a mac

Comment (MuriloDalRi) [reply README.md]: I'll add this link: https://yarnpkg.com/en/docs/install

**Comment by samuelhwilliams** [comment]
* Upgrade `gulp-include` from v2.0.0 to v2.3.1 as v2 fails to include jQuery correctly, breaking frontend-build.
* Upgrade `frontend-toolkit` to 27.0.0 to pull in shim-links correctly, identified missing by the updated gulp-include.
* Fix `hogan.js` import path
* Update `include` to `require`, which will only import the file a maximum of once.

**Pull request merged**